applications/FluidDynamicsApplication/custom_elements/two_fluid_navier_stokes.cpp

```diff
@@ -1942,6 +1944,43 @@ void TwoFluidNavierStokes<TElementData>::load(Serializer &rSerializer)
     KRATOS_SERIALIZE_LOAD_BASE_CLASS(rSerializer, BaseType);
 }
 
+
+template <class TElementData>
+void TwoFluidNavierStokes<TElementData>::GetValueOnIntegrationPoints(   const Variable<double> &rVariable,
+                                                                        std::vector<double> &rValues,
+                                                                        const ProcessInfo &rCurrentProcessInfo )
+{
+    if (rVariable == DIVERGENCE){
+
+        auto p_geom = this->pGetGeometry();
+        const GeometryType::IntegrationPointsArrayType& IntegrationPoints = p_geom->IntegrationPoints(GeometryData::GI_GAUSS_2);
+        const unsigned int num_gauss = IntegrationPoints.size();
+
+        if (rValues.size() != num_gauss){
+            rValues.resize(num_gauss);
+        }
+
+        Vector GaussPtsJDet = ZeroVector(num_gauss);
+        GeometryData::ShapeFunctionsGradientsType DN_DX;
+        p_geom->ShapeFunctionsIntegrationPointsGradients(DN_DX, GaussPtsJDet, GeometryData::GI_GAUSS_2);
+
+        for (unsigned int i_gauss = 0; i_gauss < num_gauss; i_gauss++){
+
+            const Matrix gp_DN_DX = DN_DX[i_gauss];
+            double DVi_DXi = 0.0;
+
+            for(unsigned int nnode = 0; nnode < NumNodes; nnode++){
+
+                const Vector vel = (*p_geom)[nnode].GetSolutionStepValue( VELOCITY );
```

Comment:
```suggestion
                const Vector vel = r_geom[nnode].GetSolutionStepValue( VELOCITY );
```

Reply:
Note that the type of velocity is `array_1d<double,3>` (fixed length), not `Vector` (which is an unbounded vector), so you are actually doing a copy here.